src/main/java/edu/ucla/library/iiif/fester/verticles/S3BucketVerticle.java

```diff
@@ -0,0 +1,264 @@
+
+package edu.ucla.library.iiif.fester.verticles;
+
+import static edu.ucla.library.iiif.fester.Constants.EMPTY;
+
+import com.amazonaws.regions.RegionUtils;
+
+import info.freelibrary.util.Logger;
+import info.freelibrary.util.LoggerFactory;
+import info.freelibrary.vertx.s3.S3Client;
+import info.freelibrary.vertx.s3.UserMetadata;
+
+import edu.ucla.library.iiif.fester.Config;
+import edu.ucla.library.iiif.fester.Constants;
+import edu.ucla.library.iiif.fester.HTTP;
+import edu.ucla.library.iiif.fester.MessageCodes;
+import edu.ucla.library.iiif.fester.Op;
+import edu.ucla.library.iiif.fester.utils.CodeUtils;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Promise;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.eventbus.Message;
+import io.vertx.core.http.ConnectionPoolTooBusyException;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.shareddata.Counter;
+
+/**
+ * Stores submitted manifests to an S3 bucket.
+ */
+public class S3BucketVerticle extends AbstractFesterVerticle {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(S3BucketVerticle.class, Constants.MESSAGES);
+
+    private static final long MAX_RETRIES = 10;
+
+    private S3Client myS3Client;
+
+    /**
+     * Starts the S3 Bucket Verticle.
+     */
+    @Override
+    @SuppressWarnings({ "deprecation" })
+    public void start(final Future<Void> aFuture) {
+        getJsonConsumer().handler(message -> {
+
+            // grab the Fester configuration
+            final JsonObject config = config();
+
+            // Initialize the S3BucketVerticle by getting our s3 configs and setting up the S3 client
+            if (myS3Client == null) {
```

Comment:
Move this and the `config()` stuff outside of the `getJsonConsumer().handler(...)`. We want this inside `start()` so it's done when the verticle is started, but don't need to make the check on every message we receive.